src/main/java/org/apache/sling/feature/maven/mojos/IncludeArtifactMojo.java

```diff
@@ -119,4 +186,61 @@ private void includeArtifact(final Feature f, final String extensionName, final
         }
         container.add(art);
     }
+
+    private void installFMDescriptor(File file, Feature feature) {
+        Collection<org.apache.maven.artifact.Artifact> artifacts = Collections.synchronizedCollection(new ArrayList<>());
+        if(file.exists() && file.canRead()) {
+            getLog().debug("FM File to be installed: " + file.getAbsolutePath());
+            // Need to create a new Artifact Handler for the different extension and an Artifact to not
+            // change the module artifact
+            DefaultArtifactHandler fmArtifactHandler = new DefaultArtifactHandler("slingosgifeature");
+            ArtifactId artifactId = feature.getId();
+            DefaultArtifact fmArtifact = new DefaultArtifact(
+                artifactId.getGroupId(), artifactId.getArtifactId(), artifactId.getVersion(),
+                null, "slingosgifeature", artifactId.getClassifier(), fmArtifactHandler
+            );
+            fmArtifact.setFile(file);
+            artifacts.add(fmArtifact);
+            project.addAttachedArtifact(fmArtifact);
+        } else {
+            getLog().error("Could not find FM Descriptor File: " + file);
+        }
+    }
+
+    private void addDependencies(Feature feature) {
+        // Add Dependencies if configured so
+        for(String includeDependencyScope: includeDependenciesWithScope) {
+            List<Dependency> dependencies = project.getDependencies();
+            getLog().info("Project Dependencies: " + dependencies);
+            for(Dependency dependency: dependencies) {
+                if(includeDependencyScope.equals(dependency.getScope())) {
+                    getLog().info("Include Artifact: " + dependencies);
+                    ArtifactId id = new ArtifactId(
+                        dependency.getGroupId(), dependency.getArtifactId(), dependency.getVersion(), dependency.getClassifier(), dependency.getType()
+                    );
+                    Artifact artifact = new Artifact(id);
+                    if(bundlesStartOrder >= 0) {
+                        artifact.setStartOrder(bundlesStartOrder);
+                    }
+                    feature.getBundles().add(artifact);
+                }
+            }
+        }
+    }
+
+    private void includeFeatures(Map<String, Feature> selection, Feature feature) {
```

Comment:
This inclusion looks strange - why do we only pick extensions and framework properties? Why not bundles, configurations?

Reply:
I just wasn't sure what should be merge in. I added both bundles and configurations to the includeFeatures() method.

Reply:
The question is, do we need this at all? We have a mojo that is able to aggregate, so you could first create the aggregation with the aggregate mojo and have full control over how that is done and then you run the include mojo and add the main artifact/dependencies.
You can also do it in reverse order, first use the include mojo and then aggregate with other features.

Reply:
I could not make this work - at least not in the way with include-artifact and then aggregate because aggregate would not include the generated FM.

Reply:
Got it working. Removed the method and updated the IT tests and documentation.